contracts/contracts/rem.swap/src/rem.swap.cpp

```diff
@@ -0,0 +1,307 @@
+/**
+ *  @copyright defined in eos/LICENSE.txt
+ */
+
+#include <rem.swap/rem.swap.hpp>
+
+namespace eosio {
+
+    void swap::init( const name& user, const public_key& swap_key, const string& trx_id,
+                     const string& chain_id, const string eth_address, const asset& quantity,
```

Comment:
No need to pass chain_id to init and finish. It should be hardcoded in swap contract.